arbnode/transaction_streamer.go

```diff
@@ -1870,10 +1882,36 @@ func (s *TransactionStreamer) pollToResubmitEspressoTransactions(ctx context.Con
 	return s.espressoTxnsPollingInterval
 }
 
-func (s *TransactionStreamer) shouldSubmitEspressoTransaction() bool {
+func (s *TransactionStreamer) shouldSubmitEspressoTransaction(pos *uint64) bool {
 	if s.espressoClient == nil && s.lightClientReader == nil {
 		return false
 	}
+	if s.Brige != nil && pos != nil && s.l1Reader != nil {
+		// check if the pos is already finalized on L1
+		// and get the current finalized block number from L1
+		finalizedBlockNumber, err := s.l1Reader.LatestFinalizedBlockNr(context.Background())
+		if err != nil {
+			log.Error("failed to get finalized block number", "err", err)
+			// In case of an error, we choose the safe path and still send to espresso
+			return true
+		}
+
+		sequencerMessageCount, err := s.Brige.SequencerReportedSubMessageCount(&bind.CallOpts{
+			BlockNumber: big.NewInt(int64(finalizedBlockNumber)),
```

Comment:
I think here you need to do something like:
```suggestion
			BlockNumber: new(big.Int).SetUint64(finalizedBlockNumber),
```
or declaring something like: to use as the variable. 
```
newInt := big.NewInt(0)
newInt.SetUint64(finalizedBlockNumber)
```
That way we don't anger the linter, and this code will be valid for about twice as many block numbers.